app/controllers/admin_controller.rb

```diff
@@ -43,6 +43,21 @@ def demote_basic
     redirect_to '/profile/' + @user.username + '?_=' + Time.now.to_i.to_s
   end
 
+  def reset_user_password
+    if current_user && (current_user.role == 'moderator' || current_user.role == 'admin')
+      user = User.find(params[:id])
+      if user
+        key = user.generate_reset_key
+        user.save({})
+        # send key to user email
+        PasswordResetMailer.reset_notify(user, key) unless user.nil? # respond the same to both successes and failures; security
+      end
+      flash[:notice] = I18n.t('users_controller.password_reset_email')
+      redirect_to "/login"
```

Comment:
I think the page should not redirect to the log in page since it's the admin that's starting the password reset

Reply:
Hmm, can you detect that it's not a typical user and redirect to... the profile page?